mne_bids/write.py

```diff
@@ -932,12 +932,24 @@ def write_raw_bids(raw, bids_basename, bids_root, events_data=None,
                        "raw.times should not have changed since reading"
                        " in from the file. It may have been cropped.")
 
-    params = _parse_bids_filename(bids_basename, verbose)
-    subject_id, session_id = params['sub'], params['ses']
-    acquisition, task, run = params['acq'], params['task'], params['run']
+    # convert to BIDS Path
+    if isinstance(bids_basename, str):
+        params = _parse_bids_filename(bids_basename, verbose)
```

Comment:
(https://github.com/mne-tools/mne-bids/pull/446#discussion_r439810883)

@agramfort kept this 'if' here since it doesn't seem desirable to do the _parse_bids_filename every time, unless necessary, but if you think  it's better to also get rid of, then I can change.